pkg/collector/corechecks/network-devices/cisco-sdwan/client/client_test.go

```diff
@@ -538,3 +538,52 @@ func TestGetHardwareStates(t *testing.T) {
 	// Ensure endpoint has been called 1 times
 	require.Equal(t, 1, handler.numberOfCalls())
 }
+
+func TestGetCloudExpressMetrics(t *testing.T) {
+	timeNow = mockTimeNow
+	mux := setupCommonServerMux()
+
+	handler := newHandler(func(w http.ResponseWriter, r *http.Request, calls int32) {
+		query := r.URL.Query()
+		count := query.Get("count")
+		timeZone := query.Get("timeZone")
+		startDate := query.Get("startDate")
+		endDate := query.Get("endDate")
+
+		require.Equal(t, "2000", count)
+		require.Equal(t, "UTC", timeZone)
+		require.Equal(t, "1999-12-31T23:50:00", startDate)
+		require.Equal(t, "2000-01-01T00:00:00", endDate)
+
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(fixtures.FakePayload(fixtures.GetCloudExpressMetrics)))
+	})
+
+	mux.HandleFunc("/dataservice/data/device/statistics/cloudxstatistics", handler.Func)
+
+	server := httptest.NewServer(mux)
+	defer server.Close()
+
+	client, err := testClient(server)
+	require.NoError(t, err)
+
+	devices, err := client.GetCloudExpressMetrics()
+	require.NoError(t, err)
+
+	require.Equal(t, "10.10.1.13", devices[0].VmanageSystemIP)
+	require.Equal(t, "10.10.1.13", devices[0].LocalSystemIP)
+	require.Equal(t, "10.10.1.11", devices[0].GatewaySystemIP)
+	require.Equal(t, "mpls", devices[0].LocalColor)
+	require.Equal(t, "mpls", devices[0].RemoteColor)
+	require.Equal(t, "FALSE", devices[0].BestPath)
+	require.Equal(t, "amazon_aws", devices[0].Application)
+	require.Equal(t, "amazon-group", devices[0].NbarAppGroupName)
+	require.Equal(t, float64(1), devices[0].VpnID)
+	require.Equal(t, float64(1721819993833), devices[0].EntryTime)
+	require.Equal(t, float64(404), devices[0].Latency)
+	require.Equal(t, float64(0), devices[0].Loss)
+	require.Equal(t, "5.0", devices[0].VqeScore)
+
+	// Ensure endpoint has been called 1 times
+	require.Equal(t, 1, handler.numberOfCalls())
```

Comment:
in which cases the endpoint would be called multiple times ?

Reply:
Pagination, if for example we need to query more data points than the `maxCount` param, the client will make multiple queries to get everything

Reply:
I see 👍 
should there be a test case for that ?

Reply:
The pagination logic is shared between endpoints so I didn't write pagination tests for each specific endpoint, but common pagination tests are here: https://github.com/DataDog/datadog-agent/blob/def18e3815c6cc67d8669408032961804821687a/pkg/collector/corechecks/network-devices/cisco-sdwan/client/request_test.go#L361-L422